sale_order_line_serial_number/models/sale_order.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Quartile Limited
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import models, fields, api
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    @api.multi
+    def action_confirm(self):
+        res = super(SaleOrder, self).action_confirm()
+        for order in self:
```

Comment:
@TimLai125 This entire part is looking a bit complicated.  Shouldn't it suffice if we just override this method to assign the serial number?
https://github.com/odoo/odoo/blob/f1c72ee8ee44c35707d5a18167eb8efeca3c4a6f/addons/mrp/models/procurement.py#L57-L73

Reply:
@yostashiro  I tried to change the logic to the following method:
```procurement_order.py
    def _prepare_mo_vals(self, bom):
        res = super(ProcurementOrder, self)._prepare_mo_vals(bom)
        serial_number_list = []
        if self.group_id:
            for procurement in self.group_id.procurement_ids:
                if procurement.sale_line_id and \
                        procurement.sale_line_id.serial_number:
                    serial_number_list.append(
                        procurement.sale_line_id.serial_number)
        res['serial_number'] = ','.join(serial_number_list)
        return res
```
However, this method does not handle the multiple order lines scenario well, both all the procurement orders of the order lines will belong to the same `procurement.group`, i.e. all the `remarks` and `serial_number` will be linked.
![image](https://user-images.githubusercontent.com/27185427/55309374-2d16bf00-5490-11e9-9cf3-ed1cbfb41594.png)

In order to check which procurement order belongs to which sales order line, we need to check into the stock move of the manufacturing orders, e.g. look up the relationship between the MOs ,which does not make much different to the existing logic.

Reply:
FYI: https://github.com/rfhk/cip-custom/pull/134